room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KspTypeElement.kt

```diff
@@ -251,6 +251,38 @@ internal sealed class KspTypeElement(
         return declaration.classKind == ClassKind.OBJECT
     }
 
+    override fun isCompanionObject(): Boolean {
+        return declaration.isCompanionObject
+    }
+
+    override fun isAnnotationClass(): Boolean {
+        return declaration.classKind == ClassKind.ANNOTATION_CLASS
+    }
+
+    override fun isNormalClass(): Boolean {
+        return declaration.classKind == ClassKind.CLASS
+    }
+
+    override fun isDataClass(): Boolean {
+        return Modifier.DATA in declaration.modifiers
+    }
+
+    override fun isInlineClass(): Boolean {
+        return Modifier.INLINE in declaration.modifiers
+    }
+
+    override fun isFunctionalInterface(): Boolean {
+        // Update this once KSP supports it https://github.com/google/ksp/issues/393
+        throw UnsupportedOperationException(
```

Comment:
i'm a bit conflcited on throwing because it will break a processor just by calling this even when the answer would be false. 
on the other hand returning false without checking is not great.

maybe log an error an System.error and return false instead of crashing?